### ISSUE TITLE
Processing batches with different sequence lengths using stacked LSTM layers

### ISSUE THREAD

**Issue by gorjanradevski** [tensorflow/tensorflow#31816]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): MacOS 10.14.4
- TensorFlow installed from (source or binary):  Source
- TensorFlow version (use command below): 2.0.0-beta1
- Python version: 3.65

**Describe the current behavior**

An exception is raised when trying to stack multiple `tf.keras.layers.LSTM`, while the sequence length changes across batches. This behavior occurs if the `tf.keras.Model` is built with model subclassing. On the other hand, if the model is built using the functional API, everything works as intended.

**Describe the expected behavior**

Because of the identical implementations, besides the difference in the way the model is built (subclassing and functional API), I would expect the results to be the same. In other words, I am confused why an exception is raised at all if using model subclassing.

**Code to reproduce the issue**
```
import numpy as np
import tensorflow as tf

def train_generator():
    while True:
        sequence_length = np.random.randint(10, 100)
        x_train = np.random.random((1000, sequence_length, 5))
        y_train = np.random.random((1000, sequence_length, 2))

        yield x_train, y_train

# Works as intended
model = tf.keras.models.Sequential()
model.add(tf.keras.layers.LSTM(32, return_sequences=True, input_shape=(None, 5)))
model.add(tf.keras.layers.LSTM(8, return_sequences=True))
model.add(tf.keras.layers.Dense(2))
model.compile(optimizer="adam", loss="mse")
model.fit_generator(train_generator(), steps_per_epoch=2, epochs=2, verbose=1)

# Throws an exception
class LSTMModel(tf.keras.Model):
    def __init__(self):
        super(LSTMModel, self).__init__()
        self._lstm_0 = tf.keras.layers.LSTM(32, return_sequences=True, input_shape=(None, 5)) 
        self._lstm_1 = tf.keras.layers.LSTM(8, return_sequences=True)
        self._dense = tf.keras.layers.Dense(2)

    def call(self, inputs, training=False):
        output = self._lstm_0(inputs)
        output = self._lstm_1(output)
        output = self._dense(output)

        return output

model = LSTMModel()
model.compile(optimizer="adam", loss="mse")
model.fit_generator(train_generator(), steps_per_epoch=2, epochs=2, verbose=1)
```

**Other info / logs**
>InvalidArgumentError:  [_Derived_]  Operation expected a list with 58 elements but got a list with 88 elements.
	 [[{{node gradients/TensorArrayUnstack/TensorListFromTensor_grad/TensorListStack}}]]
	 [[Adam/gradients_24/lstm_model_22/lstm_56/StatefulPartitionedCall_grad/StatefulPartitionedCall]] [Op:__inference_keras_scratch_graph_75269]

**Comment by gadagashwini** [comment]
Was able to replicate the issue using Tensorflow 2.0.0.beta1 on Colab. Please see the gist [here](https://colab.research.google.com/drive/1yaYGJbGwKghWUCo-R0f9icTkz3Mk8Q1C). Thanks!

**Comment by jvishnuvardhan** [comment]
@gorjanradevski I cannot reproduce the issue with `!pip install tf-nightly-gpu-2.0-preview`. I think this was resolved in `!pip install tf-nightly-gpu-2.0-preview`. Please take a loot at the [gist here](https://colab.sandbox.google.com/gist/jvishnuvardhan/78ed85d5cce4286d7bd8d04247f4c819/tf_31816_fit_generator.ipynb). 

I am closing the issue. Please feel free to open if the issue persists again. Thanks!
